docs/get-started/build/quickstart/deploy.mdx

```diff
@@ -0,0 +1,253 @@
+you want your app to do, and you might have the skills to write your own contract from scratch. 
+
+For simplicity, we're going to stick with `Counter.sol`, but you can use any template you want, or 
+refer to our [contract templates section](../../tooling/contracts-templates/index.mdx) for more inspiration. 
+
+We'll use the `forge create` command to deploy the contract. First, we must get an RPC endpoint and
+ensure we have a secure method for using a private key.
+
+### RPC endpoints
+
+#### Get your RPC endpoint
+
+To deploy a contract, you need to interact with the blockchain, which means you need an [RPC  endpoint](../../tooling/node-providers/index.mdx) 
+to send those calls to. 
+
+To keep things simple, we're going to use the default public RPC endpoint for Linea Sepolia: 
+`https://rpc.sepolia.linea.build`.
+
+Public endpoints such as this one are rate limited, and if you intend to deploy your app for public
+use, you'll need an endpoint better equipped to handle the load. We recommend heading to the 
+MetaMask Developer Dashboard to get a private RPC endpoint ([instructions](https://docs.metamask.io/developer-tools/dashboard/get-started/create-api/)).
+
+You can also find alternative public and private RPC endpoints on our [node providers page](../../tooling/node-providers/index.mdx).
+
+#### Update `foundry.toml`
+
+Add the RPC endpoint to the `foundry.toml` file:
+
+```toml
+[profile.rpc-endpoints]
+sepolia = "https://rpc.sepolia.linea.build"
+```
+
+### Store your private key
+
+Deploying a contract requires an account's private key, which means you need to either provide it in
+the command line when running the deployment command (not recommended) or store it securely and 
+access it with alternative methods. See the [MetaMask support guide to finding your private key](https://support.metamask.io/managing-my-wallet/secret-recovery-phrase-and-private-keys/how-to-export-an-accounts-private-key/).
+
+:::tip[Private key security best practices]
+
+We recommend [creating a new account in MetaMask](https://support.metamask.io/managing-my-wallet/accounts-and-addresses/how-to-add-accounts-in-your-wallet/)
+specifically for this project, and [naming it](https://support.metamask.io/managing-my-wallet/accounts-and-addresses/how-do-i-change-my-account-name-/)
+something appropriate, such as "linea-project".
+
+Even though we're working on a testnet, it's best to behave as if real assets were at stake, and
+work in a sandboxed context with a somewhat expendable account.
+
+:::
+
+A standard approach is to store your private key as an environment variable in a `.env` file that 
+is only stored locally (i.e. listed in `.gitignore` file). However, we're going to use Foundry's
+keystore system to encrypt and store the private key, and then access it with a password. This 
+ensures that you won't accidentally expose your private key—for example, in a GitHub repository.
```

Comment:
This is just a suggestion, but I like to link to this evergreen resource when appropriate:

`(...) your private key—for example, [in a GitHub repository](https://consensys.io/blog/how-to-avoid-uploading-your-private-key-to-github-approaches-to-prevent-making-your-secrets-public)

Reply:
I think we'll keep the scope narrowed to the built-in Foundry keystore, which works neatly given that's already the context we're in. We also reference the .env method.